SU2_CFD/src/solvers/CIncEulerSolver.cpp

```diff
@@ -2725,217 +2222,218 @@ void CIncEulerSolver::BC_Outlet(CGeometry *geometry, CSolver **solver_container,
+    /*--- Set transport properties at the outlet. ---*/
 
-        /*--- Set the normal vector and the coordinates ---*/
+    V_outlet[nDim+4] = nodes->GetLaminarViscosity(iPoint);
+    V_outlet[nDim+5] = nodes->GetEddyViscosity(iPoint);
+    V_outlet[nDim+6] = nodes->GetThermalConductivity(iPoint);
 
-        visc_numerics->SetNormal(Normal);
-        visc_numerics->SetCoord(geometry->nodes->GetCoord(iPoint),
-                                geometry->nodes->GetCoord(Point_Normal));
+    /*--- Set the normal vector and the coordinates ---*/
 
-        /*--- Primitive variables, and gradient ---*/
+    visc_numerics->SetNormal(Normal);
+    visc_numerics->SetCoord(geometry->nodes->GetCoord(iPoint),
+                            geometry->nodes->GetCoord(Point_Normal));
 
-        visc_numerics->SetPrimitive(V_domain, V_outlet);
-        visc_numerics->SetPrimVarGradient(nodes->GetGradient_Primitive(iPoint),
-                                          nodes->GetGradient_Primitive(iPoint));
+    /*--- Primitive variables, and gradient ---*/
 
-        /*--- Turbulent kinetic energy ---*/
+    visc_numerics->SetPrimitive(V_domain, V_outlet);
+    visc_numerics->SetPrimVarGradient(nodes->GetGradient_Primitive(iPoint),
+                                      nodes->GetGradient_Primitive(iPoint));
 
-        if ((config->GetKind_Turb_Model() == SST) || (config->GetKind_Turb_Model() == SST_SUST))
-          visc_numerics->SetTurbKineticEnergy(solver_container[TURB_SOL]->GetNodes()->GetSolution(iPoint,0),
-                                              solver_container[TURB_SOL]->GetNodes()->GetSolution(iPoint,0));
+    /*--- Turbulent kinetic energy ---*/
 
-        /*--- Compute and update residual ---*/
+    if ((config->GetKind_Turb_Model() == SST) || (config->GetKind_Turb_Model() == SST_SUST))
+      visc_numerics->SetTurbKineticEnergy(solver_container[TURB_SOL]->GetNodes()->GetSolution(iPoint,0),
+                                          solver_container[TURB_SOL]->GetNodes()->GetSolution(iPoint,0));
 
-        auto residual = visc_numerics->ComputeResidual(config);
+    /*--- Compute and update residual ---*/
 
-        LinSysRes.SubtractBlock(iPoint, residual);
+    auto residual_v = visc_numerics->ComputeResidual(config);
 
-        /*--- Jacobian contribution for implicit integration ---*/
-        if (implicit)
-          Jacobian.SubtractBlock2Diag(iPoint, residual.jacobian_i);
+    LinSysRes.SubtractBlock(iPoint, residual_v);
 
-      }
+    /*--- Jacobian contribution for implicit integration ---*/
+    if (implicit)
+      Jacobian.SubtractBlock2Diag(iPoint, residual_v.jacobian_i);
 
-    }
   }
 
-  /*--- Free locally allocated memory ---*/
-  delete [] Normal;
-
 }
 
 void CIncEulerSolver::SetResidual_DualTime(CGeometry *geometry, CSolver **solver_container, CConfig *config,
                                         unsigned short iRKStep, unsigned short iMesh, unsigned short RunTime_EqSystem) {
```

Comment:
It is a bit more complicated because of the "if energy" and the conversion from primitive to conservative variables

Reply:
👍 ok I saw the `V2U` stuff and wondered whether that could be generalized in some way, but if the energy stuff comes on top it might not make too much sense

Reply:
Yeah it changes whether the loops run to nVar or to nVar-1... It could be done but it felt convoluted.